### PR TITLE
feat(gatsby): distinguish total number of pages from number of written html files

### PR DESCRIPTION
**Pull request by pieh**
## Description

Make a distinction between total number of pages versus number of html files to write 

## Related Issues

[ch23149]

### CODE REVIEW

File: packages/gatsby-telemetry/src/telemetry.ts [hunk 1]
```diff
@@ -475,7 +475,10 @@ export class AnalyticsTracker {
     this.metadataCache[event] = Object.assign(cached, obj)
   }
 
-  addSiteMeasurement(event: string, obj): void {
+  addSiteMeasurement(
+    event: string,
+    obj: ITelemetryTagsPayload["siteMeasurements"]
+  ): void {
```
Comment (pieh) [inline packages/gatsby-telemetry/src/telemetry.ts:481]: I can revert this part - just though some extra type safety won't hurt

**Review by wardpeet — APPROVED** [review]
Looks good!

File: packages/gatsby-telemetry/src/telemetry.ts (hunk 1 above)
Comment (jamo) [reply packages/gatsby-telemetry/src/telemetry.ts:481]: This is great!

**Review by jamo — APPROVED** [review]
Amazing

**Review by wardpeet — APPROVED** [review]

**Pull request merged**